### PR TITLE
PE-Bear like interface is supported in the Section Widget

### PR DESCRIPTION
**Pull request by Vane11ope**
Section Widget has got PE-bear like interface. See the gif below.
gifrecord_2018-11-11_072855

### CODE REVIEW

**Comment by Maijin** [comment]
@Vane11ope such great work! 

Would you be able to add the address of the current offset "0xXXX" in both graphs and the section name probably in red color too? (even if it's dup information, I think it could help improve readibility for those cases where the sections are big)

**Comment by xarkes** [comment]
If I can add another suggestion: Could it be possible to cap a section section?
For example when it's bigger than x, just set maximum height to y, so the user doesn't have to scroll down too much to reach next section.
If it's not easy to do right now, it can be done in another PR later.

**Comment by Vane11ope** [comment]
@xarkes can and have thought, also "mouse click and seek" too etc. but i think it should be separated or it's gonna be too huge for one PR.

**Comment by Vane11ope** [comment]
improvement
@Maijin are you happy now or? :)

**Comment by Maijin** [comment]
yep looks good to me 👍

File: src/widgets/SectionsWidget.cpp [hunk 1]
```diff
@@ -191,3 +221,129 @@ void SectionsWidget::onSectionsDoubleClicked(const QModelIndex &index)
...
+
+SectionAddrDock::SectionAddrDock(SectionsModel *model, AddrType type, QWidget *parent) :
+    QDockWidget(parent),
+    header(new QLabel),
+    graphicsScene(new QGraphicsScene),
+    graphicsView(new QGraphicsView)
+{
+    switch (type) {
+        case SectionAddrDock::Raw:
+            header->setText("Raw");
```
Comment (xarkes) [inline src/widgets/SectionsWidget.cpp]: Add `tr()`

File: src/widgets/SectionsWidget.cpp [hunk 2]
```diff
@@ -191,3 +221,129 @@ void SectionsWidget::onSectionsDoubleClicked(const QModelIndex &index)
...
+            break;
+        case SectionAddrDock::Virtual:
+            header->setText("Virtual");
```
Comment (xarkes) [inline src/widgets/SectionsWidget.cpp]: Add `tr()`

File: src/widgets/SectionsWidget.cpp [hunk 3]
```diff
@@ -191,3 +221,129 @@ void SectionsWidget::onSectionsDoubleClicked(const QModelIndex &index)
...
+{
+    RVA offset = Core()->getOffset();
+    float padding = targetDock->nameHeightMap[name] * ratio;
+    int y = targetDock->namePosYMap[name] + (int)padding;
+    QGraphicsRectItem *indicator = new QGraphicsRectItem(QRectF(0, y, indicatorWidth, indicatorHeight));
+    indicator->setBrush(QBrush(Qt::red));
+    targetDock->graphicsScene->addItem(indicator);
+
+    targetDock->addTextItem(Qt::red, QPoint(targetDock->rectOffset + targetDock->rectWidth, y - indicatorParamPosY), name);
+    targetDock->addTextItem(Qt::red, QPoint(0, y - indicatorParamPosY), QString("0x%1").arg(offset, 0, 16));
```
Comment (xarkes) [inline src/widgets/SectionsWidget.cpp]: Use `ConfigColor()` and find the best color for that.

**Comment by Vane11ope** [comment]
done for all.

**Comment by Vane11ope** [comment]
Applied @ITAYC0HEN comments  "Indicator should move along with the seek changes".

**Pull request merged**